src/Avalonia.Base/Collections/AvaloniaList.cs

```diff
@@ -480,12 +480,16 @@ public virtual bool Remove(T item)
         public virtual void RemoveAll(IEnumerable<T> items)
         {
             Contract.Requires<ArgumentNullException>(items != null);
-
+            
+            var list = new List<T>();
```

Comment:
It would be beneficial to create a benchmark for this. I wonder if we should presize `list` if we know that `items` is an `IList` with a known size.

Reply:
Why do we need `IEnumerable<T>` here can't we use `IList<T>` instead. No need for type checks.

Reply:
>  I wonder if we should presize list if we know that items is an IList with a known size.

 I can't garantee that all deleting items are exists in `_inner`. So I can't presize `list` anyway

Reply:
> Why do we need `IEnumerable<T>` here can't we use `IList<T>` instead. No need for type checks.

I think this may be useful in some cases. For example we can use **linq** result as `items` parameter without calling `ToList()` method